src/Microsoft.ComponentDetection.Detectors/nuget/NuGetProjectModelProjectCentricComponentDetector.cs

```diff
@@ -264,19 +103,27 @@ private void NavigateAndRegister(
         ISingleFileComponentRecorder singleFileComponentRecorder,
         LockFileTargetLibrary library,
         string parentComponentId,
-        HashSet<string> dotnetRuntimePackageNames,
+        FrameworkPackages frameworkPackages,
         HashSet<string> visited = null)
     {
-        if (this.IsAFrameworkComponent(dotnetRuntimePackageNames, library.Name, library.Dependencies)
-            || library.Type == ProjectDependencyType)
+        if (library.Type == ProjectDependencyType)
         {
             return;
         }
 
+        var isFrameworkComponent = frameworkPackages.IsAFrameworkComponent(library.Name, library.Version);
+        var isDevelopmentDependency = IsADevelopmentDependency(library);
+
         visited ??= [];
 
         var libraryComponent = new DetectedComponent(new NuGetComponent(library.Name, library.Version.ToNormalizedString()));
-        singleFileComponentRecorder.RegisterUsage(libraryComponent, explicitlyReferencedComponentIds.Contains(libraryComponent.Component.Id), parentComponentId);
+        singleFileComponentRecorder.RegisterUsage(libraryComponent, explicitlyReferencedComponentIds.Contains(libraryComponent.Component.Id), parentComponentId, isDevelopmentDependency: isFrameworkComponent || isDevelopmentDependency);
```

Comment:
`isDevelopmentDependency` is used to flag dependencies that are never shipped with binaries to production, is that the case for all the framework components? If someone introduced the dependency directly in their project it shouldn't be treated as Development dependency right?

Reply:
> is that the case for all the framework components

Yes, that's the case.  This is meant to represent the cases where the .NET SDK will drop the package assets in favor of the framework assets.

> If someone introduced the dependency directly in their project it shouldn't be treated as Development dependency right?

No, this would still be dropped by the SDK.  For example if I have a `net8.0` project and directly reference `System.Text.Json` with version <= `8.0.0` it will not be included in my app.